support nbd-log

Pating opened the pull request:
Signed-off-by: Xie Changlong <xiechanglong@cmss.chinamobile.com>

xiubli commented on cli/nbd-cli.c:
```diff
@@ -118,11 +118,11 @@ static int load_our_module(void)
         switch (state) {
         case KMOD_MODULE_BUILTIN:
-            nbd_out("Module '%s' is builtin\n", kmod_module_get_name(mod));
+            nbd_info("Module '%s' is builtin\n", kmod_module_get_name(mod));
```
Let's make this as DEBUG info, and the same as the followings.

Pating replied:
OK

xiubli commented on daemon/nbd-runner.c:
```diff
@@ -130,8 +130,8 @@ static void *nbd_map_svc_thread_start(void *arg)
-        nbd_out("bind on port %d failed, %s\n", iport, strerror(errno));
-        nbd_out("will try to use port %d!\n", ++iport);
+        nbd_info("bind on port %d failed, %s\n", iport, strerror(errno));
+        nbd_info("will try to use port %d!\n", ++iport);
```
It will be better as WARN messages here.

Pating replied:
OK

xiubli commented on daemon/nbd-svc-routines.c:
```diff
@@ -289,7 +289,7 @@ static int nbd_parse_from_json_config_file(void)
             strcpy(dev->bstore, key);
 
-            nbd_out("key: %s, type: %d, nbd: %s, maptime: %s, size: %ld, blksize: %ld, prealloc: %d, readonly: %d\n",
+            nbd_info("key: %s, type: %d, nbd: %s, maptime: %s, size: %ld, blksize: %ld, prealloc: %d, readonly: %d\n",
```
Please make this as DEBUG messages.

Pating replied:
OK

xiubli commented on utils/nbd-log.h:
```diff
@@ -28,59 +28,48 @@
+int nbd_setup_log(char *log_dir);
+void nbd_destroy_log(void);
+int nbd_make_absolute_logfile(char *path, const char *filename);
+int nbd_resetup_log_file(char *log_dir);
```
Please remove this useless API.

Pating replied:
OK

xiubli commented on utils/nbd-log.h:
```diff
@@ -28,59 +28,48 @@
+unsigned int nbd_get_log_level(void);
+int nbd_setup_log(char *log_dir);
+void nbd_destroy_log(void);
+int nbd_make_absolute_logfile(char *path, const char *filename);
```
Please remove this and make it as static in the nbd-log.c file.

Pating replied:
OK

xiubli commented on daemon/gluster.c:
```diff
@@ -35,7 +35,7 @@
 #include "nbd-common.h"
 
-#define NBD_GFAPI_LOG_FILE "/var/log/nbd-runner.log"
+#define NBD_GFAPI_LOG_FILE "/var/log/nbd-runner-glfs.log"
```
Please include the "nbd-sysconfig.h" header file and then change this line as:
```
#define NBD_GFAPI_LOG_FILE NBD_LOG_DIR_DEFAULT"/nbd-runner-glfs.log"
```

xiubli commented on utils/strlcpy.h:
```diff
@@ -0,0 +1,8 @@
+
+#include <sys/types.h>
+
+size_t strlcpy(char *dst, const char *src, size_t size);
```
And also let switch all the strncpy to strlcpy at the same time in the following files:
cli/nbd-cli-cmd.c
daemon/gluster.c
daemon/nbd-svc-routines.c

And comment it here for the reason:

For strncpy there is one warning:
Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated. The strlcpy code comes from the BSD and tcmu-runner project.

Pating replied:
OK

Commit:
utils: fix spelling mistake

1. TIPRC -> TIRPC
2. fix indent issue

Signed-off-by: Xie Changlong <xiechanglong@cmss.chinamobile.com>

Commit:
switch to strlcpy instead of strncpy

For strncpy there is one warning:
Warning: If there is no null byte among the first n bytes of src,
the string placed in dest will not be null-terminated.

The strlcpy code comes from the BSD and tcmu-runner project.

Signed-off-by: Xie Changlong <xiechanglong@cmss.chinamobile.com>

Commit:
support nbd log

Signed-off-by: Xie Changlong <xiechanglong@cmss.chinamobile.com>
